### ISSUE TITLE
Remove Unused Import from Groovy Files

### ISSUE THREAD

**Issue by PoonamGargAgarwal** [diffplug/spotless#245]
If you are submitting a **bug**, please include the following:

- [ ] summary of problem
- [ ] gradle or maven version
- [ ] spotless version
- [ ] operating system and version
- [ ] copy-paste your full Spotless configuration block(s), and a link to a public git repo that reproduces the problem if possible
- [ ] copy-paste the full content of any console errors emitted by `gradlew spotless[Apply/Check] --stacktrace`

If you're just submitting a feature request or question, no need for the above.

**Comment by PoonamGargAgarwal** [comment]
Problem Summary : I would like to remove unused import from groovy file.

**Comment by fvgh** [comment]
I am not sure whether this is a real formatter task. You cannot check wildcard imports since the formatter only cares about a single file.
If it is implemented, I would propose to provided it for Java and Groovy, since there should be not much difference.
Not sure whether you are aware, that CodeNarc handles this issue very well:
http://codenarc.sourceforge.net/codenarc-rules-imports.html

@nedtwigg 
I see that there is a demand and many people consider wild card imports anyway bad practice and can with the means of Spotless forbid wild cards. So in principal this feature might be nice for Groovy and Java, but I am not sure whether we should leave it to CodeNarc and FindBugs, since these tools are prepared to handle the topic completely.

**Comment by nedtwigg** [comment]
We can (and do) remove unused imports from java files if that file doesn't have any wildcard imports.  I think it's fine to have a groovy formatter that says `if (hasWildcards) { return } else { removeUnusedImports() }`.

I also think it would be okay to have a separate "no wildcard imports" rule, but it's a better fit for CodeNarc etc. than Spotless, because there's no way to implement it as `F(unformatted) = formatted`.  If somebody wrote a "no wildcard imports" rule, I'd be happy to merge it.

**Comment by PoonamGargAgarwal** [comment]
@nedtwigg Yes i do agree codenarc has rules for removing unused imports for Groovy files. But it doesn't delete line of unused import like the diffspotless plugin works for java file. Codenarc rules lets  user know about violation of rule (unused import).  My Question is IF we have groovy file & has unused import , lets our spotless plugin identify & delete from groovy file. I hope this makes sense.
